pyproject.toml

```diff
@@ -21,8 +21,8 @@ classifiers = [
 ]
 dynamic = ["version"]
 dependencies = [
-    "pydantic>=2.0",
-    "pydantic-settings>=2.0",
+    "pydantic>=2.7",
+    "pydantic-settings>=2.4",
```

Comment:
Needed because accessing the current state requires `pydantic-settings>=2.4` which in turn requires `pydantic>=2.7` hence I raised that as well.